src/redux/slices/pick-pane-slice.ts

```diff
@@ -0,0 +1,206 @@
+      action: PayloadAction<{
+        pickPaneSetName: string;
+        id: string;
+      }>
+    ) => {
+      const arg = action.payload;
+      const sets = state.pickPaneSets[arg.pickPaneSetName];
+      const element = sets?.pickPanes.find((item) => item.id === arg.id);
+      // "custom" element only can be deleted.
+      if (element && element.custom) {
+        sets.pickPanes = sets.pickPanes.filter((item) => item.id !== arg.id);
+        if (sets.pickedPickPaneId === arg.id) {
+          sets.pickedPickPaneId = "";
+        }
+      }
+    },
+  },
+
+  extraReducers: (builder) => {
+    builder
+      .addCase(
+        addPickPane.fulfilled,
+        (
+          state: IPickPaneState,
+          action: PayloadAction<{
+            pickPaneSetName: string;
+            pickPane: IPickPane;
+            picked: string | null;
+            index: number | undefined;
+          }>
+        ) => {
+          const arg = action.payload;
+          if (!state.pickPaneSets[arg.pickPaneSetName]) {
+            state.pickPaneSets[arg.pickPaneSetName] = {
+              pickPanes: [],
+              pickedPickPaneId: "",
+            };
+          }
+          const element =
+            arg.pickPane.id &&
+            state.pickPaneSets[arg.pickPaneSetName].pickPanes.find(
+              (item) => item.id === arg.pickPane.id
+            );
+          if (!element) {
+            if (typeof arg.index === "number") {
+              state.pickPaneSets[arg.pickPaneSetName].pickPanes.splice(
+                arg.index,
+                0,
+                arg.pickPane
+              );
+            } else {
+              state.pickPaneSets[arg.pickPaneSetName].pickPanes.push(
+                arg.pickPane
+              );
+            }
+          }
+          if (arg.picked) {
+            state.pickPaneSets[arg.pickPaneSetName].pickedPickPaneId =
+              arg.pickPane.id;
+          }
+        }
+      )
+      .addCase(
+        setPickPanePicked.fulfilled,
+        (
+          state: IPickPaneState,
+          action: PayloadAction<{
+            pickPaneSetName: string;
+            pickPane: IPickPane | null;
+          }>
+        ) => {
+          const arg = action.payload;
+          state.pickPaneSets[arg.pickPaneSetName].pickedPickPaneId =
+            arg.pickPane?.id || "";
+        }
+      );
+  },
+});
+
+export const addPickPane = createAsyncThunk<
+  {
+    pickPaneSetName: string;
+    pickPane: IPickPane;
+    picked: string | null;
+    index: number | undefined;
+  },
+  {
+    pickPaneSetName: string;
+    pickPane: IPickPane;
+    setCurrent: boolean;
+    index?: number;
+  }
+>("addPickPane", async ({ pickPaneSetName, pickPane, setCurrent, index }) => {
+  // if setCurrent === true, we need to fetch the content
+  // if setCurrent === false, it's enough to fetch the icon only.
+  // The content will be fetched when the user pick a pane.
+  await pickPane.fetchPickPane(setCurrent);
```

Comment:
Fetch logic should not be in this slice. The component should initiate loading by url when the selected texture has been changed. As the slice is generic, fetch is not needed for basemaps etc.